Notes

baha-kenpachi opened the pull request:
консольное приложение с несколькими экранами
экран просмотра архивов
экран создания архивов
экран просмотра заметок в архиве
экран создание заметки с добавлением содержания
экран просмотра выбранной заметки

Commit:
осталось дописать NoteMenu.kt
дописать Note.kt
После того как программа будет готово, планируется переписать его немного и добавить в работу согласно подсказке "Для всего этого советуем использовать изменяемый список, который содержит в себе название пункта и лямбду, которая вызовется при выборе этого пункта."

Commit:
Программа работает,
осталось добавить в работу согласно подсказке "Для всего этого советуем использовать изменяемый список, который содержит в себе название пункта и лямбду, которая вызовется при выборе этого пункта."

Commit:
немного изменил способ хранения данных

Commit:
теперь показывает содержание заметки

vlad-zotov commented on src/main/kotlin/ArchiveMenu.kt:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    override fun handleInput(input: String) {
+
```
пустые строки помогают улучшать читаемость кода, но не в этом случае. Рекомендую удалить её.

baha-kenpachi replied:
> пустые строки помогают улучшать читаемость кода, но не в этом случае. Рекомендую удалить её.

удаляю все пустые строки и при нажатии option+command+L IDE автоматом проставляет тут пустую строку

vlad-zotov replied:
понял. В таком случае здесь всё ок!

vlad-zotov commented on src/main/kotlin/ArchiveMenu.kt:
```diff
@@ -0,0 +1,64 @@
+import model.Archive
+
+class ArchiveMenu : Screen(null) {
```
часто применяется подход в именовании, когда в имени класса указывается название класса, от которого он наследуется. Предлагаю переименовать класс в `ArchiveScreen`, так будет понятнее с каким объектом мы работаем

vlad-zotov commented on src/main/kotlin/ArchiveMenu.kt:
```diff
@@ -0,0 +1,64 @@
+
+        val selected =
+            menu.keys.find { // через find проходим по каждому ключу с 0 до 1 индекса ключа если данное значение совпадает с input то возвращаем ключ и присваеваем к selected
+                it.substring(0, 1) == input
```
тебе не нужно брать подстроку от ключа, таким образом ты не сможешь зайти в 10й элемент. Попробуй сделать 10 архивов и получить доступ к 10 :) Это нужно поправить.

vlad-zotov commented on src/main/kotlin/NoteMenu.kt:
```diff
@@ -0,0 +1,65 @@
+    override fun updateMenu() {
+        menu = mutableMapOf()
+        menu.put("0. Создать заметку", { showCreateNoteScreen() })
+        //val notes = Data.getArchiveByName(archiveName!!)?.notes
```
лишние комментарии можно впоследствии почистить

vlad-zotov commented on src/main/kotlin/NoteMenu.kt:
```diff
@@ -0,0 +1,65 @@
+        }
+    }
+
+    override fun showMenu() { // функция выводит через итерацию каждую строку списка
```
в каждой функции `showMenu` у тебя есть методы 
```
updateMenu()
menu.forEach { t, u -> println(t) }
```
можешь вынести их в класс родителя, таким образом ты уменьшишь число повторяемого кода!

baha-kenpachi committed:
Update Readme.md

Commit:
снято ограничение на добавление более 9 архивов и заметок :)
убрал лишние комментарии и подровнял весь код
переименовал ArchiveMenu на ArchiveScreen так-как этот клас наследуется Screen.kt, так же и с NoteMenu.kt на NoteScreen.kt

Commit:
снято ограничение на добавление более 9 архивов и заметок :)
убрал лишние комментарии и подровнял весь код
переименовал ArchiveMenu на ArchiveScreen так-как этот клас наследуется Screen.kt, так же и с NoteMenu.kt на NoteScreen.kt
